test/unit/org/apache/cassandra/cql3/validation/entities/VirtualTableTest.java

```diff
@@ -317,6 +365,32 @@ public void testModifications() throws Throwable
                              "INSERT INTO test_virtual_ks.vt2 (key, value) VALUES ('pk2', 22) IF NOT EXISTS");
     }
 
+    @Test
+    public void testTruncationOnTruncatableTable() throws Throwable
+    {
+        // check for existing records
+        assertRows(execute("SELECT * FROM test_virtual_ks.vt4"),
+                row("pk1", 1),
+                row("pk2", 2));
+
+        // truncate
+        execute("TRUNCATE test_virtual_ks.vt4");
+
+        // check that truncation is applied
+        assertEmpty(execute("SELECT * FROM test_virtual_ks.vt4"));
+    }
+
+    @Test
+    public void testInvalidDMLOperations() throws Throwable
```

Comment:
I moved truncation test from `testInvalidDDLOperations` to `testInvalidDMLOperations` because `TRUNCATE` is a DML operation. Also I added a test for `INSERT` into read-only VT - it was missing.